### PR TITLE
✨ Source Mongodb POC: Configurable Snapshot Fetch Size

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Make the fetch size for the initial snapshot configurable

## How
* Add the fetch size to the specification
* Use the current value (1000) as the default if not set

## Recommended reading order
1. `spec.json`
2. `MongoDbSource.java`

### CODE REVIEW

**Review by colesnodgrass — APPROVED** [review]

File: airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/MongoDbStateIterator.java [hunk 1]
```diff
@@ -104,7 +111,18 @@ public AirbyteMessage next() {
...
+  private boolean shouldContinue() {
+    if (count > 0) {
+      if (batchSize > 0) {
+        return count % batchSize == 0;
+      } else {
+        return true;
+      }
+    } else {
+      return false;
+    }
```
Comment (colesnodgrass) [inline airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/MongoDbStateIterator.java:126]: This isn't a `should continue` method but rather a `should emit state` method.  

This change introduces a bug where no state will be emitted if `batchSize = 0`.

With the previous implementation, the `batchSize` and the `emit state every N messages` were coupled together.  I wonder if we should decouple this now if we want to support a `batchSize = 0`.

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`92288f2ae5`](https://github.com/airbytehq/airbyte/pull/29433/commits/92288f2ae5f2ca21542b2510d7a48fce0fab15b7)) - ❌

⏲️  Total pipeline duration: 06mn47s 

| Step                                                                                   | Result   |
|:---------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-internal-poc/metadata.yaml     | ✅        |
| Connector version semver check                                                         | ✅        |
| Connector version increment check                                                      | ❌        |
| QA checks                                                                              | ❌        |
| Build connector tar                                                                    | ✅        |
| Build source-mongodb-internal-poc docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mongodb-internal-poc:integrationTest | ✅        |
| Acceptance tests                                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_configure-fetch-size/1692115492/92288f2ae5f2ca21542b2510d7a48fce0fab15b7/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:92288f2ae5f2ca21542b2510d7a48fce0fab15b7)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

File: airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/MongoDbStateIterator.java (hunk 1 above)
Comment (jdpgrailsdev) [reply airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/MongoDbStateIterator.java:126]: @colesnodgrass Thanks for the clarification.  I'm not really sure how to decouple it, other than creating a different implementation to select when the batch size is 0.  Any other suggestions?

Comment (colesnodgrass) [reply airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/MongoDbStateIterator.java:126]: I think for now if `batchSize == 0` we should default it to some value (500? 1000?) and revisit this decision in the future.

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`37fa6ca6fe`](https://github.com/airbytehq/airbyte/pull/29433/commits/37fa6ca6fec587001b0408c0626e514443d0afab)) - ❌

⏲️  Total pipeline duration: 05mn31s 

| Step                                                                                   | Result   |
|:---------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-internal-poc/metadata.yaml     | ✅        |
| Connector version semver check                                                         | ✅        |
| Connector version increment check                                                      | ❌        |
| QA checks                                                                              | ❌        |
| Build connector tar                                                                    | ✅        |
| Build source-mongodb-internal-poc docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mongodb-internal-poc:integrationTest | ✅        |
| Acceptance tests                                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_configure-fetch-size/1692116970/37fa6ca6fec587001b0408c0626e514443d0afab/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:37fa6ca6fec587001b0408c0626e514443d0afab)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-approvington** [comment]
After a careful ML study, <br/> we think this looks okay.  <br/> imagine code being okay

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**